app/src/main/java/com/example/albumsapp/AlbumsListFragment.kt

```diff
@@ -33,13 +35,27 @@ class AlbumsListFragment : Fragment() {
         recyclerView = binding.album
         recyclerView.layoutManager = LinearLayoutManager(context)
 
-        viewModel.albums.observe(viewLifecycleOwner) {
-            recyclerView.adapter = AlbumAdapter(it, ::onAlbumClicked)
+        viewModel.albums.observe(viewLifecycleOwner) { event ->
+            if (event != null) {
+                binding.album.adapter = AlbumAdapter(event, ::onAlbumClicked)
+            } else {
+                viewModel.errorLiveData.value?.let { showErrorAlertDialog(it) }
```

Comment:
let's make it simpler: in `onViewCreated` just observe `viewModel.errorLiveData` and show error in its body. 

```
viewModel.errorLiveData.observe(viewLifecycleOwner) {
    showErrorAlertDialog(it)
}
```